src/main/java/com/nha/abdm/wrapper/hip/hrp/dataTransfer/DataTransferService.java

```diff
@@ -1,51 +1,327 @@
+import java.util.UUID;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
 import org.springframework.stereotype.Service;
+import reactor.core.Exceptions;
 
 @Service
 public class DataTransferService implements DataTransferInterface {
-  /**
-   * The callback from ABDM gateway after consentGrant for dataTransfer,
-   * POST method for /on-notify
-   * @param HIPConsentNotification careContext and demographics details are provided, and implement a logic to check the existence of the careContexts.
-   */
-  @Override
-  public void notifyOnReceived(HIPConsentNotification HIPConsentNotification) {
-    // TODO Implement onNotify workflow logic.
+  private static final Logger log = LogManager.getLogger(DataTransferService.class);
+
+  @Value("${facilityUrl}")
+  public String facilityUrlPath;
+
+  @Value("${dataOnRequestPath}")
+  public String dataOnRequestPath;
+
+  @Value("${dataPushNotificationPath}")
+  public String dataPushNotificationPath;
+
+  @Autowired PatientRepo patientRepo;
+  @Autowired LogsRepo logsRepo;
+  private ErrorResponse errorResponse = new ErrorResponse();
+  private final RequestManager requestManager;
+  private final HIPClient hipClient;
+  @Autowired RequestLogService requestLogService;
+  @Autowired PatientService patientService;
+  @Autowired EncryptionController encryptionController;
+
+  @Autowired
+  public DataTransferService(HIPClient hipClient, RequestManager requestManager) {
+    this.hipClient = hipClient;
+    this.requestManager = requestManager;
   }
 
   /**
-   *
    * POST /on-request as an acknowledgement for agreeing to make dataTransfer to ABDM gateway.
-   * @param HIPHealthInformationRequest HIU public keys and dataPush URL is provided
+   *
+   * @param hipHealthInformationRequest HIU public keys and dataPush URL is provided
    */
   @Override
-  public void requestOnReceived(HIPHealthInformationRequest HIPHealthInformationRequest) {
-    // TODO Implement onRequest workflow logic.
-    initiateBundleRequest(HIPHealthInformationRequest);
+  public void requestOnReceived(HIPHealthInformationRequest hipHealthInformationRequest)
+      throws IllegalDataStateException {
+    HIPHealthInformationRequestAcknowledgement hipHealthInformationRequestAcknowledgement = null;
+    RespRequest responseRequestId =
+        RespRequest.builder().requestId(hipHealthInformationRequest.getRequestId()).build();
+    String consentId = hipHealthInformationRequest.getHiRequest().getConsent().getId();
+    HiRequest hiRequest =
+        HiRequest.builder()
+            .sessionStatus("ACKNOWLEDGED")
+            .transactionId(hipHealthInformationRequest.getTransactionId())
+            .build();
+    if (logsRepo.findByConsentId(consentId) != null) {
```

Comment:
this is not right. we should not look for consent in request-logs repo but rather in patients collection